tdi_python/tdiTable.py

```diff
@@ -0,0 +1,2617 @@
+        return count.value
+
+    def get_capacity(self):
+        count = c_size_t(0)
+        session = self._cintf.get_session()
+        dev_target = self._cintf.get_dev_tgt()
+        sts = self._cintf.tdi_table_size_get(self._handle, session, dev_target, byref(count))
+        if not sts == 0:
+            raise TdiTableError("Error: get capacity failed on table {}. [{}]".format(self.name, self._cintf.err_str(sts)), self, sts)
+        return count.value
+
+    def get_type(self):
+        table_type = c_int(-1)
+        sts = self._cintf.get_driver().tdi_table_type_get(self._handle, byref(table_type))
+        #print("table name= "+str(self.name)+" table_type="+str(table_type))
+        print("table_type=0x%x" % table_type.value)
+        if not sts == 0:
+            raise TdiTableError("Error: get table_type failed on table {}. [{}]".format(self.name, self._cintf.err_str(sts)), self, sts)
+        return table_type.value
+
+    def set_supported_attributes_to_supported_commands(self):
+        num_attrs = c_uint(0)
+        sts = self._cintf.get_driver().tdi_table_num_attributes_supported(self._handle, byref(num_attrs))
+        if sts != 0:
+            raise TdiTableError("Error: num attributes supported get failed on table {}. [{}]".format(self.name, self._cintf.err_str(sts)), self, sts)
+        arr_type = c_uint * num_attrs.value
+        attributes_arr = arr_type()
+        sts = self._cintf.get_driver().tdi_table_attributes_supported(self._handle, byref(attributes_arr), byref(num_attrs))
+        if sts != 0:
+            raise TdiTableError("Error: attributes supported get failed on table {}. [{}]".format(self.name, self._cintf.err_str(sts)), self, sts)
+        for i in range(len(attributes_arr)):
+            if attributes_arr[i] == 0:
+                self.supported_commands.append("symmetric_mode_set")
+                self.supported_commands.append("symmetric_mode_get")
+            elif attributes_arr[i] == 2:
+                self.supported_commands.append("idle_table_set_poll")
+                self.supported_commands.append("idle_table_set_notify")
+                self.supported_commands.append("idle_table_get")
+            elif attributes_arr[i] == 3:
+                self.supported_commands.append("dynamic_hash_set")
+                self.supported_commands.append("dynamic_hash_get")
+            elif attributes_arr[i] == 4:
+                self.supported_commands.append("meter_byte_count_adjust_set")
+                self.supported_commands.append("meter_byte_count_adjust_get")
+            elif attributes_arr[i] == 5:
+                self.supported_commands.append("port_status_notif_cb_set")
+            elif attributes_arr[i] == 6:
+                self.supported_commands.append("port_stats_poll_intv_set")
+                self.supported_commands.append("port_stats_poll_intv_get")
+            elif attributes_arr[i] == 8:
```

Comment:
We need to come up with a way to provide different target drivers to be able to add commands which are target specific easily and remove these from here

Reply:
Yes. We will do it in the next phase per our discussion.